python/cudf/cudf/core/frame.py

```diff
@@ -2028,6 +2028,11 @@ def sum(
         b    34
         dtype: int64
         """
+        if isinstance(self, cudf.BaseIndex):
+            warnings.warn(
+                "Support for Index.sum is deprecated and will be removed",
```

Comment:
Would it be better to do 
```suggestion
                f"Support for {self.__class__}.sum is deprecated and will be removed",
```
here and throughout?